test/src/server.py

```diff
@@ -0,0 +1,1558 @@
+
+        Parameters
+        ----------
+        net_type : str
+            Test interface network type
+        machine_type : str
+            Guest machine type
+        root_disk : str
+            Path to the disk file for guest's root partition
+        debug_qemu : bool
+            True if you want to attach GDB to Qemu. The GDB server will
+            be bound to port 1234.
+        ioregionfd : bool
+            True if you want to use the IORegionFD enhanced virtio_net_device
+            for the test interface.
+        qemu_build_dir : str
+            Path to the Qemu build directory. Can be empty if you want to use
+            the installed Qemu.
+        vhost : bool
+            True if you want to use vhost on the test interface.
+        rx_queue_size : int
+            Size of the receive queue for the test interface.
+        tx_queue_size : int
+            Size of the transmit queue for the test interface.
+
+        Returns
+        -------
+        """
+        # TODO this command should be build by the Guest object
+        # it should take all the settings from the config file
+        # and compile them.
+        dev_type = 'pci' if machine_type == 'pc' else 'device'
+        test_net_config = (
+            f" -netdev tap,vhost={'on' if vhost else 'off'}," +
+            f'id=admin1,ifname={self.test_tap},script=no,' +
+            'downscript=no,queues=4' +
+            f' -device virtio-net-{dev_type},id=testif,' +
+            f'netdev=admin1,mac={self.guest_test_iface_mac},mq=on' +
+            (',use-ioregionfd=true' if ioregionfd else '')
+            + f',rx_queue_size={rx_queue_size},tx_queue_size={tx_queue_size}'
+            # + f',rx_queue_size={rx_queue_size},tx_queue_size={tx_queue_size}'
+            # f' -device virtio-net-{dev_type},id=testif,' +
+            # ' -device rtl8139,id=testif,' +
+            # 'netdev=admin1,mac=52:54:00:fa:00:60,mq=on' +
+        ) if net_type == 'brtap' else (
+            f" -netdev tap,vhost={'on' if vhost else 'off'}," +
+            'id=admin1,fd=3 3<>/dev/tap$(cat ' +
+            f'/sys/class/net/{self.test_macvtap}/ifindex) ' +
+            f' -device virtio-net-{dev_type},id=testif,' +
+            'netdev=admin1,mac=$(cat ' +
+            f'/sys/class/net/{self.test_macvtap}/address)' +
+            (',use-ioregionfd=true' if ioregionfd else '')
+            + f',rx_queue_size={rx_queue_size},tx_queue_size={tx_queue_size}'
+            # + f',rx_queue_size={rx_queue_size},tx_queue_size={tx_queue_size}'
+            # f' -device virtio-net-{dev_type},id=testif,' +
+            # ' -device rtl8139,id=testif,' +
+        )
+        qemu_bin_path = 'qemu-system-x86_64'
+        if qemu_build_dir:
+            qemu_bin_path = path_join(qemu_build_dir, qemu_bin_path)
+        disk_path = self.guest_root_disk_path
+        if root_disk:
+            disk_path = root_disk
+        home = "/scratch/okelmann" # TODO make config var in autotest.cfg
```

Comment:
this needs to be configurable

Reply:
Resolved in 222b1a3d03a50fc66a65522a9405db71aefce5b2 ... see commit message for more details, tldr: set for example fsdevs=home:/home/gierens,nixstore:/nix/store in your autotest.cfg to passthrough /home/gierens as fsdev with mount tag home and /nix/store with mount tag nixstore.